### PR TITLE
Implemented draw line and erase functionality

### PR DESCRIPTION
**Pull request by charleswitherspoon**

### CODE REVIEW

**Comment by charleswitherspoon** [comment]
This pull request covers tasks 9.1.1 and 9.1.2 (draw and erase tools, respectively). It also covers task task 18.1 (selecting an element).

**Comment by charleswitherspoon** [comment]
This pull request also implements tasks 10.1, 10.2, and 10.3 which all cover undo and redo functionality.

**Comment by charleswitherspoon** [comment]
Removed superfluous code and added more documentation.

File: src/main/java/com/groupseven/pdfproject/PageModel.java [hunk 1]
```diff
@@ -38,11 +38,18 @@ public PageModel(BufferedImage image) {
...
         
-        graphicsContext = canvas.getGraphicsContext();
+        graphicsContext = canvas.getCanvas().getGraphicsContext2D();
         
         node = new VBox(0);
+        BackgroundImage backgroundImage = new BackgroundImage(
+                fximage,
+                BackgroundRepeat.NO_REPEAT,
+                BackgroundRepeat.NO_REPEAT,
+                BackgroundPosition.CENTER,
```
Comment (DryCreations) [inline src/main/java/com/groupseven/pdfproject/PageModel.java:50]: Using bp center causes issues when resizing the window, the drawing on the canvas is aligned to the left and top, while the image is now centered.

To make them consistent you can create a new left aligned background posititon:

```java
BackgroundPosition bp = new BackgroundPosition(Side.LEFT, 0, false, Side.TOP, 0, false);
BackgroundImage backgroundImage = new BackgroundImage(
                fximage,
                BackgroundRepeat.NO_REPEAT,
                BackgroundRepeat.NO_REPEAT,
                bp,
                BackgroundSize.DEFAULT);
```

**Review by DryCreations — APPROVED** [review]

**Review by leonard-parker5 — APPROVED** [review]
I tested on my system to confirm that the tools were working properly, and that the issues mentioned earlier were corrected for. Nothing code-wise stands out at the moment as needing improvements. Nice job, I think the task has been accurately implemented.

Drawing, erasing, creating rectangles, and the undo/redo gave me no issues and worked appropriately.

Untitled

**Review by Cassandra-Mae-Patel — APPROVED** [review]
Great Efforts!
This let's me successfully draw  lines and rectangles and also it let's me to erase it. Undo and redo also works great. 
Just the few things which I find odd is that, if I draw a line on a rectangle as shown in the following image and then after selecting the rectangle followed by mouse drag, the line disappears form the rectangle. What is happening is that, the rectangle is brought in front, in other words, line is sent a layer back (behind the rectangle). Then, on pressing the undo won't change anything because I presume controls are in new layer now.

1] line on rectangle:
image

2]What happens on drag:
image

3] Changes not affected on undo/redo:
image

Except that everything works smoothly.  Hence, approved.

**Pull request merged**